### PR TITLE
Add photoswipe zoom support

### LINKED ISSUE #34: Add Photoswipe zoom support to image component
Add an optional parameter that enables click-to-zoom support using Photoswipe.

### PR DESCRIPTION
**Pull request by anita-chouhan**
Fixes #34

### CODE REVIEW

File: assets/css/main.css [hunk 1]
```diff
@@ -5,3 +5,4 @@
 @import 'components';
 @import 'tailwindcss/utilities';
 @import 'utilities';
+@import 'photoswipe';
```
Comment (rungta) [inline assets/css/main.css]: Can this not be replaced with `@import 'photoswipe/dist/photoswipe.css';` directly? No need for an extra file?

File: assets/js/photoswipeinit.esm.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Init Photoswipe.js
+// es6
```
Comment (rungta) [inline assets/js/photoswipeinit.esm.js:2]: Add a link to the Photoswipe library / docs here in the comments.

File: assets/js/photoswipeinit.esm.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  if (!lightbox) {
+    lightbox = import(window.PHOTOZOOM.photoswipe.lightbox).then(
+      (module) => module.default
+    );
+  }
+  return lightbox;
+}
+
+function initPhotoswipe() {
+  if (document.querySelector(options.gallery)) {
```
Comment (rungta) [inline assets/js/photoswipeinit.esm.js]: We should do a sanity check for `window.PHOTOZOOM.*` variables to be declared before proceeding.

File: templates/_layouts/html.twig [hunk 4]
```diff
@@ -69,6 +69,21 @@
...
       
+      {% block script_inline %}
+        {# JS Constants #}
+        {% set jsConstants = {
+          photoswipe: {
+            lightbox: 'photoswipe-lightbox.min.js',
+            module: 'photoswipe.min.js',
+          }|map((href, key) => alias('@assetBaseUrl/build/' ~ href|replace('/.(css|js)/', ".#{assetManifest[href]}.$1"))),
+        }|merge(jsConstants ?? {})|filter %}
+        <div id="pjax-eval">
```
Comment (rungta) [inline templates/_layouts/html.twig:80]: Any significance of the `id="pjax-eval"`? Do we have to add this to the pjax selectors list?

**Review by rungta — CHANGES_REQUESTED** [review]

File: assets/css/main.css (hunk 1 above)
Comment (anita-chouhan) [reply assets/css/main.css]: Yes `photoswipe.css` file is not needed. Added direct import to `main.css`

File: assets/js/photoswipeinit.esm.js (hunk 2 above)
Comment (anita-chouhan) [reply assets/js/photoswipeinit.esm.js:2]: Done.

File: assets/js/photoswipeinit.esm.js (hunk 3 above)
Comment (anita-chouhan) [reply assets/js/photoswipeinit.esm.js]: Done.

File: templates/_layouts/html.twig (hunk 4 above)
Comment (anita-chouhan) [reply templates/_layouts/html.twig:80]: Added to pjax selectors list after checking the same usage in _Vahura_ [code](https://github.com/miranj/vahura/blob/a1e25c873512c967f57c6b87af33d960a6f26fd2/assets/js/pjaxinit.js#L14).

**Review by rungta — APPROVED** [review]

**Pull request merged**